### ISSUE TITLE
Needs to Support Cookies

### ISSUE THREAD

**Issue by JamesWiresmith** [ni/niauth-js#1]
During the login post you have a commented out line "credentials: "same-origin"". (https://github.com/ni/niauth-js/blob/master/index.js#L189)

This means that the additional services aren't authenticated.

You obviously reference on the first GET that this causes other issues with prior logins which need resolving but I don't see any other way of using this library with additional web service calls

**Comment by JamesWiresmith** [comment]
Looks like the login 200 response returns the logged in user in the body. I get a response:
Logged in as: admin

I think the correct function would be:
* Allow the cookies.
* If you get a 200 to the parse out the username.
* If it matches. do nothing more.
* If it doesnt then logout and re-login as the new user.

An API call to check this on application load would also be fantastic. How does this fit with your use case?

**Comment by bstreiff** [comment]
Yep, that hadn't been fully baked. Commit 7b777f28baac should resolve most of what you're asking for. `updateFromSession` is the "check on application load" function you describe.

The only thing that's missing is the automatic "logout-and-re-login" case; the current implementation of the "login" function is starting to get a bit unwieldy, so I'd like to have a crack at splitting that up first before I add yet another callback-indentation level to deal with automatic logout-and-back-in.
